docs/tutorials/on-device-training/ios-app.md

```diff
@@ -0,0 +1,987 @@
+                
+                Spacer()
+                
+                ZStack(alignment: .center) {
+                    Image(systemName: "mic.fill")
+                        .resizable()
+                        .aspectRatio(contentMode: .fit)
+                        .frame(width: 100, height: 100)
+                        .foregroundColor( readyToRecord ? .gray: .red)
+                        .transition(.scale)
+                        .animation(.easeIn, value: 1)
+                }
+                
+                Spacer()
+                
+                Button(action: {
+                    readyToRecord = false
+                    recordVoice()
+                }) {
+                    Text(readyToRecord ? "Record" : "Recording ...")
+                        .font(.title)
+                        .padding()
+                        .background(readyToRecord ? .green : .gray)
+                        .foregroundColor(.white)
+                        .cornerRadius(10)
+                }.disabled(!readyToRecord)
+                    
+            case .trainingInProgress:
+                VStack {
+                    Spacer()
+                    ProgressView(value: trainingProgress,
+                                 total: 1.0,
+                                 label: {Text("Training")},
+                                 currentValueLabel: {Text(String(format: "%.0f%%", trainingProgress * 100))})
+                    .padding()
+                    Spacer()
+                }
+                    
+            case .trainingComplete:
+                Spacer()
+                Text("Training successfully finished!")
+                    .font(.title)
+                    .padding()
+                    .multilineTextAlignment(.center)
+                    .fontDesign(.monospaced)
+                
+                Spacer()
+                NavigationLink(destination: InferView()) {
+                    Text("Infer")
+                        .font(.title)
+                        .padding()
+                        .background(.purple)
+                        .foregroundColor(.white)
+                        .cornerRadius(10)
+                }
+                .padding(.leading, 20)
+            }
+            
+            Spacer()
+        }
+        .padding()
+        .navigationTitle("Train")
+    }
+}
+
+struct TrainView_Previews: PreviewProvider {
+    static var previews: some View {
+        TrainView()
+    }
+}
+```
+
+The complete implementation of the `TrainingView` can be found [here](https://github.com/microsoft/onnxruntime-training-examples/blob/master/on_device_training/mobile/ios/MyVoice/TrainView.swift)
+
+### Inference View
+Lastly, we will create `InferView` that will be used to perform inference with the trained model. It will prompt the user to record their voice and perform inference with the trained model. Then, it will display the result of the inference.
```

Comment:
```suggestion
Lastly, we will create the `InferView` that will be used to perform inference with the trained model. It will prompt the user to record their voice and perform inference with the trained model. Then, it will display the result of the inference.
```